executor/set_test.go

```diff
@@ -774,6 +774,11 @@ func TestGetSetNoopVars(t *testing.T) {
 	tk.MustQuery("SHOW WARNINGS").Check(testkit.Rows("Warning 8144 setting query_cache_type has no effect in TiDB"))
 	// but the change is still effective.
 	tk.MustQuery("SELECT @@query_cache_type").Check(testkit.Rows("OFF"))
+
+	// Only ON and OFF supported
+	err := tk.ExecToErr("SET GLOBAL tidb_enable_noop_variables = 2")
+	require.Error(t, err)
+	require.Equal(t, "[variable:1231]Variable 'tidb_enable_noop_variables' can't be set to the value of '2'", err.Error())
```

Comment:
Also please include a test for setting it to 'warn', which is the reported issue.

Reply:
Done. But I think it is not very necessary. When set `SET GLOBAL tidb_enable_noop_variables = 2` it means only 0 (OFF) and 1 (ON) are valid values, so it can grantee that all other values are rejected.